tests/backend/test_nongaussian_gates.py

```diff
@@ -18,13 +18,49 @@
 
 import numpy as np
 
-KAPPAS = np.linspace(0, 2 * np.pi, 7)
+from scipy.linalg import expm
 
+KAPPAS = np.linspace(0, 2 * np.pi, 7)
+GAMMAS = np.linspace(0, 2 * np.pi, 7)
 
 @pytest.mark.backends("fock", "tf")
 class TestFockRepresentation:
     """Tests that make use of the Fock basis representation."""
 
+    @pytest.mark.parametrize("gamma", GAMMAS)
+    def test_cubic_phase(self, setup_backend, gamma, cutoff, tol):
+        """Tests if the Cubic phase gate has the right effect on states in the Fock basis"""
+
+        backend = setup_backend(1)
+
+        backend.prepare_ket_state(np.ones([cutoff]) / np.sqrt(cutoff), 0)
+        backend.cubic_phase(gamma, 0)
+        s = backend.state()
+        if s.is_pure:
+            numer_state = s.ket()
+        else:
+            numer_state = s.dm()
+
+        #Create annihilation operator matrix
+        ladder_vals = np.arange(1, cutoff)
+        ladder_vals = np.sqrt(ladder_vals)
+        a = np.zeros([cutoff, cutoff])
+        np.fill_diagonal(a[:, 1:], ladder_vals)
```

Comment:
I'm surprised that the tests pass when constructing the ladder operators to size `cutoff x cutoff`. I would think that, due to the matrix exponentiation, there would be error in the last diagonal element of `gate`.

@co9olguy, any thoughts?

Reply:
I believe that, almost by definition, the action of a gate on a cutoff Hilbert space is necessarily generated by the same gate Hamiltonian but just reduced to the cutoff dimension. I think the error you're thinking of is exactly the numerical error we get from using a cutoff (which for cubic phase is particularly bad due to the 3rd order of the Hamiltonian).